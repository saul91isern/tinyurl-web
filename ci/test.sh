#!/bin/sh

echo "----- Install dependencies -----"
yarn
echo "----- Linter -----"
yarn test --ci

