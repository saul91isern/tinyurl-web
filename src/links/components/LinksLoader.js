import { useEffect } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { clearLinks, fetchLinks } from "../routines";

export const LinksLoader = ({ clearLinks, fetchLinks }) => {
  useEffect(() => {
    fetchLinks();
    return () => {
      clearLinks();
    };
  }, [clearLinks, fetchLinks]);
  return null;
};

LinksLoader.propTypes = {
  clearTemplates: PropTypes.func,
  fetchTemplates: PropTypes.func,
  scope: PropTypes.string,
};

export default connect(null, { clearLinks, fetchLinks })(LinksLoader);
