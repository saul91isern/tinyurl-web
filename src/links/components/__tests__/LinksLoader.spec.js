/* eslint-disable quotes */
import React from "react";
import { render } from "../../../test/render";
import { LinksLoader } from "../LinksLoader";

describe("<LinksLoader />", () => {
  it("renders LinksLoader", async () => {
    const fetchLinks = jest.fn();
    const clearLinks = jest.fn();
    const linksQuery = "foo";
    const { container } = render(
      <LinksLoader
        fetchLinks={fetchLinks}
        clearLinks={clearLinks}
        linksQuery={linksQuery}
      />,
      {}
    );

    expect(container).toMatchSnapshot();
    expect(fetchLinks).toHaveBeenCalledTimes(2);
    expect(fetchLinks.mock.calls[1][0]).toEqual(linksQuery);
  });
});
