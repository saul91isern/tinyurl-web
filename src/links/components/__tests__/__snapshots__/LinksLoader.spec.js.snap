// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinksLoader /> renders LinksLoader 1`] = `<div />`;
